### PR TITLE
Add output variables support for gha and bitrise

### PR DESCRIPTION
**Pull request by raghavharness**
@bradrydzewski Can you please review this. Thanks.

### CODE REVIEW

File: plugin/bitrise/bitrise.go [hunk 1]
```diff
@@ -6,16 +6,51 @@
...
 
+const (
+	envStoreFile = ".envstore.yml"
+)
+
 // Is returns true if the root path is a Bitrise
 // plugin repository.
-func Is(root string) bool {
-	path := filepath.Join(root, "step.yml")
+func Is(root, filename string) bool {
```
Comment (bradrydzewski) [inline plugin/bitrise/bitrise.go]: can you provide some more context to this change?  The purpose of the `Is` function is to automatically determine if a codebase is Bitrise.  Providing the `step.yml` as an input should not be needed.

File: plugin/bitrise/execer.go [hunk 2]
```diff
@@ -72,6 +73,17 @@ func (e *Execer) Exec(ctx context.Context) error {
 		}
 	}
 
+	// create the .envstore.yml file if not present
+	if !Is(e.Source, envStoreFile) {
```
Comment (bradrydzewski) [inline plugin/bitrise/execer.go]: we should use `Is` to determine if a plugin is a Bitrise plugin.  If we are already executing the Bitrise plugin, we should not need to invoke `Is`.  Perhaps this should be a new, private helper function?

File: plugin/bitrise/bitrise.go (hunk 1 above)
Comment (bradrydzewski) [reply plugin/bitrise/bitrise.go]: ok I see, you were looking to re-use some code.  In this case I would recommend the below change.

```Go
// Is returns true if the root path is a Bitrise
// plugin repository.
func Is(base) bool {
	return exists(base, "step.yml")
}

func exists(base, file)
	path := filepath.Join(base, file)
	if _, err := os.Stat(path); err == nil {
		return true
	}
	return false
}
```

Then you can use the `exists` function as a help in line 77 of `execer.go` without changing the purpose of the `Is` function.

File: plugin/bitrise/envstore.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package bitrise
+
+type EnvironmentItemModel map[string]string
+
+// EnvsSerializeModel ...
+type EnvsSerializeModel struct {
```
Comment (bradrydzewski) [inline plugin/bitrise/envstore.go]: should this be private instead of public?  do you plan to use this data object outside the package?

Comment (bradrydzewski) [reply plugin/bitrise/envstore.go]: I would recommend against using `Model` in struct names.  This is a common convention in Java, but it is not really used in Go programs.

File: plugin/bitrise/envstore.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package bitrise
+
+type EnvironmentItemModel map[string]string
```
Comment (bradrydzewski) [inline plugin/bitrise/envstore.go]: do you need to create a custom type for this?

Comment (raghavharness) [reply plugin/bitrise/envstore.go]: let me get rid of this, not really required

File: plugin/bitrise/envstore.go (hunk 3 above)
Comment (raghavharness) [reply plugin/bitrise/envstore.go]: Sure changing it.

File: plugin/bitrise/bitrise.go (hunk 1 above)
Comment (raghavharness) [reply plugin/bitrise/bitrise.go]: done

**Comment by raghavharness** [comment]
@bradrydzewski Addressed your comments.

File: plugin/github/workflow.go [hunk 5]
```diff
@@ -106,3 +111,26 @@ func prePostStep(name, envFile string) step {
 	}
 	return s
 }
+
+func getOutputVariables(prevStepId, outputfile string, output_vars []string) step {
+	skip := len(outputfile) == 0 || len(output_vars) == 0
+	cmd := ""
+	for _, output_var := range output_vars {
+		cmd += fmt.Sprintf("print(\"%s\"+\"=\"+\"${{ steps.%s.outputs.%s }}\"); ", output_var, prevStepId, output_var)
```
Comment (shubham149) [inline plugin/github/workflow.go:119]: this will fail for mac

Comment (shubham149) [reply plugin/github/workflow.go:119]: use camelcase

**Pull request merged**